### PR TITLE
Fix misleading exception message, cleanup code FileSystemWatcher

### LINKED ISSUE #24817: FileSystemWatcher misleading exception message
I encountered an exception being thrown in an ASP.NET Core 2.0 project which said "The directory name {0} is invalid." and it was misleading. [`FileSystemWatcher`](https://github.com/dotnet/corefx/blob/724ddb8fc95a5b1f639cfa508a32b9a8c50c1949/src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.cs#L108) throws an exception with this message, and in my case the path was valid but the directory did not exist.

"The directory name {0} does not exist." would be a more helpful message for me.

### PR DESCRIPTION
**Pull request by MarcoRossignoli**
@danmosemsft  found another place where message could be more precise again on FileSystemWatcher. Fixes #24817

### CODE REVIEW

File: src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.cs [hunk 1]
```diff
@@ -260,6 +260,7 @@ private byte[] AllocateBuffer()
         /// <devdoc>
         ///    Gets or sets the path of the directory to watch.
         /// </devdoc>
+        ///
```
Comment (stephentoub) [inline src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.cs]: Why did you add this?

File: src/System.IO.FileSystem.Watcher/src/Resources/Strings.resx [hunk 2]
```diff
@@ -127,7 +127,7 @@
     <value>Too many changes at once in directory:{0}.</value>
   </data>
   <data name="InvalidDirName" xml:space="preserve">
-    <value>The directory name {0} is invalid.</value>
+    <value>The directory name '{0}' is invalid.</value>
```
Comment (stephentoub) [inline src/System.IO.FileSystem.Watcher/src/Resources/Strings.resx]: Does this same error string appear in other projects? If it does, we should change those as well you remain consistent.

File: src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.cs [hunk 3]
```diff
@@ -271,11 +272,12 @@ public string Path
                 value = (value == null) ? string.Empty : value;
                 if (!string.Equals(_directory, value, PathInternal.StringComparison))
                 {
-                    if (!Directory.Exists(value))
-                    {
-                        throw new ArgumentException(SR.Format(SR.InvalidDirName, value));
-                    }
+                    if (value.Length == 0)
+                        throw new ArgumentException(SR.Format(SR.InvalidDirName, value), nameof(this.Path));
```
Comment (stephentoub) [inline src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.cs]: Why does this check need to be separated out?

File: src/System.IO.FileSystem.Watcher/src/Resources/Strings.resx (hunk 2 above)
Comment (danmosemsft) [reply src/System.IO.FileSystem.Watcher/src/Resources/Strings.resx]: No. Although we are pretty inconsistent with {0} vs '{0}' in general.

File: src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.cs (hunk 3 above)
Comment (danmosemsft) [reply src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.cs]: So he can use a different message.

Comment (danmosemsft) [inline src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.cs]: do you need `this.` on these?

File: src/System.IO.FileSystem.Watcher/tests/FileSystemWatcher.unit.cs [hunk 4]
```diff
@@ -528,13 +528,13 @@ public void FileSystemWatcher_Path()
 
             // FSW starts with String.Empty and will ignore setting this if it is already set,
             // but if you set it after some other valid string has been set it will throw.
-            AssertExtensions.Throws<ArgumentException>(null, () => watcher.Path = String.Empty);
+            AssertExtensions.Throws<ArgumentException>("Path", () => watcher.Path = String.Empty);
```
Comment (danmosemsft) [inline src/System.IO.FileSystem.Watcher/tests/FileSystemWatcher.unit.cs]: this is failing when run against desktop framework. you need to use this overload:
`public static void Throws<T>(string netCoreParamName, string netFxParamName, Action action)`

File: src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.cs (hunk 3 above)
Comment (stephentoub) [reply src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.cs]: I don't understand why a different message is needed. Why is empty string more special as an invalid name than others that Exists returns false for, in particular for ones that are similarly invalid like those that contain invalid for system characters? I don't see this change actually improving understandability. What an I missing?

File: src/System.IO.FileSystem.Watcher/src/Resources/Strings.resx (hunk 2 above)
Comment (stephentoub) [reply src/System.IO.FileSystem.Watcher/src/Resources/Strings.resx]: > Although we are pretty inconsistent with {0} vs '{0}' in general.

For file paths? Sounds like something we should be consistent on.

**Comment by stephentoub** [comment]
I'm still not clear on https://github.com/dotnet/corefx/pull/25485#discussion_r153037227.

**Comment by karelz** [comment]
@stephentoub sorry, I missed that one + I thought you're OOF :(

**Comment by stephentoub** [comment]
I'm here :)
